crates/accelerate/src/utils.rs

```diff
@@ -0,0 +1,41 @@
+use pyo3::prelude::*;
+
+use faer::Faer;
+use faer::IntoFaerComplex;
+use num_complex::Complex;
+use numpy::PyReadonlyArray2;
+
+/// Return indices that sort data.
+/// If `data` contains two elements that are incomparable,
+/// an error will be thrown.
+pub fn argsort<T: PartialOrd>(data: &[T]) -> Vec<usize> {
+    let mut indices = (0..data.len()).collect::<Vec<_>>();
+    indices.sort_by(|&a, &b| data[a].partial_cmp(&data[b]).unwrap());
+    indices
+}
+
+// TODO: Use traits and parameters
+/// Modulo operation. This should give the same result
+/// as `numpy.mod`.
+pub fn modulo(a: f64, b: f64) -> f64 {
```

Comment:
Not that I expect a drastically different result here, but I'd mark this as inline just so the compiler always tries to inline it:

```suggestion
#[inline]
pub fn modulo(a: f64, b: f64) -> f64 {
```